### ISSUE TITLE
Cannot instantiate class: cgeo.geocaching.apps.navi.NavigationSelectionActionProvider

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#11251]
This error shows up when refreshing a cache using a developer build based on current `release` branch. Did not check with `master`.
No crash associated.

```
2021-07-17 12:02:56.361 24690-24690/cgeo.geocaching W/SupportMenuInflater: Cannot instantiate class: cgeo.geocaching.apps.navi.NavigationSelectionActionProvider
    java.lang.NoSuchMethodException: cgeo.geocaching.apps.navi.NavigationSelectionActionProvider.<init> [class android.content.Context]
        at java.lang.Class.getConstructor0(Class.java:2363)
        at java.lang.Class.getConstructor(Class.java:1759)
        at androidx.appcompat.view.SupportMenuInflater$MenuState.newInstance(SupportMenuInflater.java:550)
        at androidx.appcompat.view.SupportMenuInflater$MenuState.readItem(SupportMenuInflater.java:425)
        at androidx.appcompat.view.SupportMenuInflater.parseMenu(SupportMenuInflater.java:179)
        at androidx.appcompat.view.SupportMenuInflater.inflate(SupportMenuInflater.java:129)
        at cgeo.geocaching.CacheMenuHandler.addMenuItems(CacheMenuHandler.java:132)
        at cgeo.geocaching.CacheMenuHandler.addMenuItems(CacheMenuHandler.java:137)
        at cgeo.geocaching.CacheDetailActivity.onCreateOptionsMenu(CacheDetailActivity.java:645)
        at android.app.Activity.onCreatePanelMenu(Activity.java:4277)
        at androidx.fragment.app.FragmentActivity.onCreatePanelMenu(FragmentActivity.java:287)
        at androidx.appcompat.view.WindowCallbackWrapper.onCreatePanelMenu(WindowCallbackWrapper.java:94)
        at androidx.appcompat.app.AppCompatDelegateImpl$AppCompatWindowCallback.onCreatePanelMenu(AppCompatDelegateImpl.java:3084)
        at androidx.appcompat.app.AppCompatDelegateImpl.preparePanel(AppCompatDelegateImpl.java:1907)
        at androidx.appcompat.app.AppCompatDelegateImpl.doInvalidatePanelMenu(AppCompatDelegateImpl.java:2188)
        at androidx.appcompat.app.AppCompatDelegateImpl$2.run(AppCompatDelegateImpl.java:268)
        at android.view.Choreographer$CallbackRecord.run(Choreographer.java:1037)
        at android.view.Choreographer.doCallbacks(Choreographer.java:845)
        at android.view.Choreographer.doFrame(Choreographer.java:775)
        at android.view.Choreographer$FrameDisplayEventReceiver.run(Choreographer.java:1022)
        at android.os.Handler.handleCallback(Handler.java:938)
        at android.os.Handler.dispatchMessage(Handler.java:99)
        at android.os.Looper.loopOnce(Looper.java:201)
        at android.os.Looper.loop(Looper.java:288)
        at android.app.ActivityThread.main(ActivityThread.java:7829)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:548)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:982)
```

**Comment by fm-sys** [comment]
I think this could be the cause for  #11216.

**Comment by moving-bits** [comment]
I can reproduce this warning in the emulator, and it would pretty much explain the error described in #11216.

Question is: Why is that warning occurring / why can't the class be found? 

The only SO info I could find for this was about a missing proguard rule which lead to this class being excluded. But we do have a specific proguard rule for NavigationSelectionActionProvider class already...

**Comment by eddiemuc** [comment]
This is really easy to explain: constructor needs to take a single parameter of type `android.content.Context`, but we define only a constructor taking a parameter of type `android.app.Activity`.

Once I changed the type to Context, the call worked. But there's a small problem: the context passed is REALLY a Context (not an Activity), so we can't cast it back. I will see whether I can make this class work also with a "pure" context

**Comment by eddiemuc** [comment]
Fixed with PR #11397
